### ISSUE TITLE
uia2_device_port_guard prevents appium from starting

### ISSUE THREAD

**Issue by kazzmir** [appium/appium#14753]
## The problem

With Appium 1.18.0 and above, the appium-uiautomator2-driver creates a /tmp/uia2_device_port_guard file in order for multiple appium processes to allocate a port for communicating with uiautomator2 on android devices without conflicting with each other. If that file already exists then appium will hang for a while on startup, and eventually return an error `EEXIST: file already exists, open '/tmp/uia2_device_port_guard'`.

This is made worse by the fact that if I pass the systemPort to appium that the port allocation process in appium-uiautomator2-driver should not be needed at all, and thus does not have to get a file lock.

This fails, but should work fine due to passing systemPort:
```
$ touch /tmp/uia2_device_port_guard
$ appium ...
$ curl -X POST http://localhost:<appium-port>  '{"capabilities": {'systemPort": 2000}}'
```

It looks like this code could check the `this.opts.systemPort` before entering the guard block.

https://github.com/appium/appium-uiautomator2-driver/blob/6ab64020cc1c9ba05aadf044775db60bcc4a4871/lib/driver.js#L276

I see the forward to adb is also contained inside the guard, but not sure if thats really needed.

For now my workaround is simply to delete the /tmp/uia2_device_port_guard just before starting appium.

**Comment by mykola-mokhnach** [comment]
Duplicate of https://github.com/appium/appium/issues/14682

**Comment by mykola-mokhnach** [comment]
The change has been published to appium@beta. Now no locking is applied if `systemPort` is provided
